src/main.ts

```diff
@@ -0,0 +1,178 @@
+import * as child_process from 'child_process'
+import * as fs from 'fs'
+import * as path from 'path'
+
+import * as ts from 'typescript'
+import * as yargs from 'yargs'
+
+import { Indexer } from './Indexer'
+import * as lsif from './lsif'
+
+export const lsif_typed = lsif.lib.codeintel.lsif_typed
+
+export interface Options {
+  projectRoot: string
+  project: string
+  writeIndex: (index: lsif.lib.codeintel.lsif_typed.Index) => void
+}
+
+export function main(): void {
+  // eslint-disable-next-line @typescript-eslint/no-floating-promises, no-unused-expressions
+  yargs
+    .scriptName('lsif-node')
+    .usage('$0 <cmd> [args]')
+    .command(
+      'index [project]',
+      'LSIF index a project',
+      yargs => {
+        yargs.positional('project', {
+          type: 'string',
+          default: '.',
+          describe: 'the directory to index',
+        })
+        yargs.option('useYarnWorkspaces', {
+          type: 'boolean',
+          default: 'false',
+          describe: 'whether to index all yarn workspaces',
+        })
```

Comment:
"use" doesn't seem appropriate here. There's already the verb `index` in the doc comment, how about reusing that, so something like `indexAllWorkspaces`/`indexYarnWorkspaces`/`indexAllYarnWorkspaces`.